src/osd/ReplicatedPG.cc

```diff
@@ -2444,7 +2444,7 @@ ReplicatedPG::cache_result_t ReplicatedPG::maybe_handle_cache_detail(
     }
 
     if (must_promote || (!hit_set && !op->need_skip_promote())) {
-      promote_object(obc, missing_oid, oloc, op, promote_obc);
+      promote_object(obc, missing_oid, oloc, op->may_write(), op, promote_obc);
```

Comment:
@zhaochao Please keep PR title < 50 characters
https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits